### PR TITLE
add flux density unit when image file is moment 0

### LINKED ISSUE #1632: missing unit for statistic = flux density
Reported by a user via the Helpdesk. 

If the input image is a moment 0 image with the pixel unit as Jy/beam km/s, after defining a region, flux density can be derived. The unit should be "Jy km/s" as the per beam part is integrated out. 

Screen Shot 2021-08-18 at 11 12 46 AM

### PR DESCRIPTION
**Pull request by acdo2002**
Fixed #1632 
I add if statement when the image file is moment 0 and then update the unitString for flux density.

### CODE REVIEW

**Comment by acdo2002** [comment]
I unify the unit for all different styles from different data cubes.
"Janky" => "Jy"; The first character uses upper case, others use lower case.

**Comment by kswang1029** [comment]
> I unify the unit for all different styles from different data cubes.
> "Janky" => "Jy"; The first character uses upper case, others use lower case.

we better not change the format as defined in the header. 
Screen Shot 2021-09-16 at 12 18 24 PM

mJy and Mjy are different things.

**Comment by acdo2002** [comment]
Keep the format as defined in the header.
I add three conditions based on the request:
"JY/BEAM" => "JY"
"jy/beam" => "jy"
"Jy/Beam" => "Jy"

**Comment by acdo2002** [comment]
I modify the method to extract the unit.
please check it, now it should even fine with mJy unit

File: src/components/Stats/StatsComponent.tsx [hunk 1]
```diff
@@ -163,8 +163,10 @@ export class StatsComponent extends React.Component<WidgetProps> {
                 } else if (type === CARTA.StatsType.SumSq) {
                     unitString = `(${unit})^2`;
                 } else if (type === CARTA.StatsType.FluxDensity) {
-                    if (unit === "Jy/beam") {
-                        unitString = "Jy";
+                    if (unit.toLowerCase().includes("beam") && !unit.toLowerCase().includes("km/s")) {
```
Comment (veggiesaurus) [inline src/components/Stats/StatsComponent.tsx]: might be more elegant to use regex here?

Comment (acdo2002) [reply src/components/Stats/StatsComponent.tsx]: I change to the regex way to find the matching string.

Comment (veggiesaurus) [reply src/components/Stats/StatsComponent.tsx]: I was thinking of doing it in one line with regex and groups. If you use separate `match` calls and `!!`, then it's simpler to just use `includes`.

**Comment by veggiesaurus** [comment]
is this logic correct? for example, if the unit is `km/s`, then the resultant string is `km.km/s` :thinking:

File: src/components/Stats/StatsComponent.tsx (hunk 1 above)
Comment (acdo2002) [reply src/components/Stats/StatsComponent.tsx]: Should I move back to the previous method: using unit.toLowerCase().include("beam")?

**Comment by acdo2002** [comment]
the original unit = Jy/beam.km/s
I use unit.split("/")[0] + ".km/s"
then the displayed unitString would be "Jy.km/s" (because I only extract the part before "/")

**Comment by veggiesaurus** [comment]
> the original unit = Jy/beam.km/s
> I use unit.split("/")[0] + ".km/s"
> then the displayed unitString would be "Jy.km/s" (because I only extract the part before "/")

I mean I tested the above code, with `unit = "km/s"` and it results in `km.km/s`.

File: src/components/Stats/StatsComponent.tsx (hunk 1 above)
Comment (confluence) [reply src/components/Stats/StatsComponent.tsx]: _Edit: I moved this to the review comment._

File: src/components/Stats/StatsComponent.tsx [hunk 2]
```diff
@@ -163,8 +163,10 @@ export class StatsComponent extends React.Component<WidgetProps> {
                 } else if (type === CARTA.StatsType.SumSq) {
                     unitString = `(${unit})^2`;
                 } else if (type === CARTA.StatsType.FluxDensity) {
-                    if (unit === "Jy/beam") {
-                        unitString = "Jy";
+                    if (/beam/gi.test(unit) && !/km\/s/gi.test(unit)) {
+                        unitString = unit.split("/")[0];
+                    } else if (/beam.km\/s/gi.test(unit)) {
+                        unitString = unit.split("/")[0] + ".km/s";
```
Comment (confluence) [inline src/components/Stats/StatsComponent.tsx]: You can replace this with `unitString = unit.replace(/\/beam/i, "");`

**Review by confluence — CHANGES_REQUESTED** [review]

**Review by confluence — APPROVED** [review]

**Review by veggiesaurus — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]

**Pull request merged**